src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,101 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int START_ITEMS_NUMBER = 10;
+    private static final double INCREASE_RATE = 1.5;
+    private T[] item;
+    private int size;
+
+    public ArrayList() {
+        item = (T[]) new Object[START_ITEMS_NUMBER];
+    }
+
     @Override
     public void add(T value) {
-
+        grow();
```

Comment:
```suggestion
        if (size == items.length) {
            grow();
        }
```
Better check if you actually need to increase the capacity, because the name of the method suggests that it will be increased without any checks

Reply:
I moved the check to the method itself - so as not to duplicate the code!